kairos-crypto/src/lib.rs

```diff
@@ -0,0 +1,20 @@
+pub mod error;
+pub mod implementations;
+
+use std::path::Path;
+
+use error::CryptoError;
+
+pub trait CryptoSigner {
+    fn from_private_key_file<P: AsRef<Path>>(file: P) -> Result<Self, CryptoError>
+    where
+        Self: Sized;
+    fn from_public_key(bytes: &[u8]) -> Result<Self, CryptoError>
+    where
+        Self: Sized;
+
+    fn sign(&self, data: &[u8]) -> Result<Vec<u8>, CryptoError>;
+    fn verify(&self, data: &[u8], signature_bytes: &[u8]) -> Result<(), CryptoError>;
```

Comment:
I'd prefer `AsRef<[u8]>` instead of `&[u8]`.

Reply:
Fixed in https://github.com/cspr-rad/kairos/pull/35/commits/9792ee33cc357f3798cca652a76e89cc4ff51755.